plugins/python/redisgears_python.c

```diff
@@ -1746,6 +1751,550 @@ static PyObject* atomicExit(PyObject *self, PyObject *args){
+    verifyRedisAILoaded();
+    if (RedisAI_DAGAddTensorGet == NULL) {
+        PyErr_SetString(GearsError, "DAG run is not supported in RedisAI version");
+        return NULL;
+    }
+    PyDAGRunner* pyDag = (PyDAGRunner*)self;
+    if(!PyObject_IsInstance((PyObject*)pyDag, (PyObject*)&PyDAGRunnerType)){
+        PyErr_SetString(GearsError, "Given argument is not of type DAGRunner");
+        return NULL;
+    }
+    if(PyTuple_Size(args) != 1){
+        PyErr_SetString(GearsError, "Wrong number of args to TensorSet op");
+        return NULL;
+    }
+    if(pyDag->dag == NULL){
+        PyErr_SetString(GearsError, "PyDAGRunner is invalid");
+        return NULL;
+    }
+    PyObject* tensorName = PyTuple_GetItem(args, 0);
+    if(!PyUnicode_Check(tensorName)){
+        PyErr_SetString(GearsError, "Tensor name argument must be a string");
+        return NULL;
+    }
+
+    const char* inputNameStr = PyUnicode_AsUTF8AndSize(tensorName, NULL);
+    RedisAI_DAGAddTensorGet(pyDag->dag, inputNameStr, NULL);
+    Py_INCREF(self);
+    return self;
+}
+
+static PyObject* appendModelRunOp(PyObject *self, PyObject *args, PyObject *kargs) {
+    verifyRedisAILoaded();
+    if (RedisAI_DAGCreateModelRunOp == NULL) {
+        PyErr_SetString(GearsError, "DAG run is not supported in RedisAI version");
+        return NULL;
+    }
+    PyDAGRunner* pyDag = (PyDAGRunner*)self;
+    if(!PyObject_IsInstance((PyObject*)pyDag, (PyObject*)&PyDAGRunnerType)){
+        PyErr_SetString(GearsError, "Given argument is not of type DAGRunner");
+        return NULL;
+    }
+    if(pyDag->dag == NULL){
+        PyErr_SetString(GearsError, "PyDAGRunner is invalid");
+        return NULL;
+    }
+
+    // Load model from keyspace and create MODELRUN op.
+    PyObject* modelName = GearsPyDict_GetItemString(kargs, "name");
+    if (!modelName) {
+        PyErr_SetString(GearsError, "Model key name was not given");
+        return NULL;
+    }
+    if(!PyUnicode_Check(modelName)){
+        PyErr_SetString(GearsError, "Model name argument must be a string");
+        return NULL;
+    }
+    const char* modelNameStr = PyUnicode_AsUTF8AndSize(modelName, NULL);
+    RedisModuleCtx *ctx = RedisModule_GetThreadSafeContext(NULL);
+    RedisGears_LockHanlderAcquire(ctx);
+    RedisModuleString* keyRedisStr = RedisModule_CreateString(ctx, modelNameStr, strlen(modelNameStr));
+    RedisModuleKey *key;
+    RAI_Error *err;
+    RedisAI_InitError(&err);
+    RAI_Model *model;
+    if(RedisAI_GetModelFromKeyspace(ctx, keyRedisStr, &key, &model, REDISMODULE_READ, err) != REDISMODULE_OK) {
+        RedisModule_FreeString(ctx, keyRedisStr);
+        RedisGears_LockHanlderRelease(ctx);
+        RedisModule_FreeThreadSafeContext(ctx);
+        PyErr_SetString(GearsError, RedisAI_GetError(err));
+        return NULL;
+    }
+
+    RedisModule_FreeString(ctx, keyRedisStr);
+    RedisGears_LockHanlderRelease(ctx);
+    RedisModule_FreeThreadSafeContext(ctx);
```

Comment:
move this into a function, for cleaner code

Reply:
done